.github/workflows/ci.yml

```diff
@@ -158,108 +101,44 @@ jobs:
-          args+=( -DCMAKE_INSTALL_PREFIX="$(pwd)/install" )
-
-          # Disable some defaults for now to speed up the build as
-          # they are not necessary in a minimal installation
-          args+=( -DRE2C_BUILD_RE2GO=OFF )
-
-          # Print the resulting command line for debugging purposes
-          printf "cmake" && printf " '%s'" "${args[@]}" && printf "\n"
-
-          # Run CMake, pass quoted arguments to prevent word split.
-          printf " '%s'" "${args[@]}" | xargs cmake
+        run: cmake --preset=${{ matrix.name }}-fast
 
       - name: Fast Build
-        run: cmake --build ${{ matrix.build-dir }} --config ${{ matrix.build-type }}
+        run: cmake --build --preset=${{ matrix.name }}-fast
 
       - name: Install
-        run: |
-          cmake --build ${{ matrix.build-dir }} --config ${{ matrix.build-type }} --target install
-          echo -e "Install tree:\n$(find install | sed 's/^/-- /')"
+        run: cmake --build --preset=${{ matrix.name }}-fast --target install
 
       - name: Minimal Install Test
         working-directory: ./install/bin
         run: ./re2c --version
 
       - name: Full Configure
-        run: |
-          # Read prepared command into an array
-          args=( ${{ steps.configure-command.outputs.cmake }} )
-
-          # Amend Build Options
-          args+=( -DCMAKE_INSTALL_PREFIX="$(pwd)/install-2" -DRE2C_BUILD_RE2GO=ON -DRE2C_BUILD_LIBS=ON )
-          args+=( -DRE2C_REBUILD_LEXERS=ON -DRE2C_FOR_BUILD="$(pwd)/install/bin/re2c" )
-
-          # Windows does not have a native 'man' and 'sed' commands.
-          # Disabled until this https://github.com/skvadrik/re2c/issues/327
-          # will be implemented
-          case ${{ runner.os }} in
-            Windows*)
-              args+=( -DRE2C_REBUILD_DOCS=OFF )
-              ;;
-            *)
-              args+=( -DRE2C_REBUILD_DOCS=ON )
-              ;;
-          esac
-
-          # Print the resulting command line for debugging purposes
-          printf "cmake" && printf " '%s'" "${args[@]}" && printf "\n"
-
-          # Run CMake, pass quoted arguments to prevent word split.
-          printf " '%s'" "${args[@]}" | xargs cmake
+        run: cmake --preset=${{ matrix.name }}-full
 
       - name: Full Build
         run: |
           find src -name '*.re' | xargs touch
-          cmake --build ${{ matrix.build-dir }} --config ${{ matrix.build-type }}
+          cmake --build --preset=${{ matrix.name }}-full
 
-      - name: Test
-        # Remove the following line after fix Windows tests
-        working-directory: ${{ matrix.build-dir }}
+      - name: Run Main Test Suite
+        run: cmake --build --preset=${{ matrix.name }}-full --target tests
+
+      - name: Get build dir
+        id: build-info
         run: |
-          # main test suite
-          cmake --build . --target check
+          build_dir="$(cmake --build --preset=${{ matrix.name }}-full --target print_bin_dir | tail -n 2 | head -n 1)"
```

Comment:
The head-tail command needs a comment to make it easier to understand.

Reply:
Bad news:

File version must be 3 or higher for condition support like this:

```json
      "condition": {
        "type": "equals",
        "lhs": "${hostSystemName}",
        "rhs": "Linux"
      }
```

Version 3 was introduced in CMake 3.21.1. 

Also `installDir` is allowed in preset files specifying version 3 or above.

Reply:
See https://github.com/skvadrik/re2c/pull/374#discussion_r687241746

Reply:
Ok, I'll mark this as resolved then.